docs/index.rst

```diff
@@ -11,10 +11,21 @@ TextAttack
 Features 
 -----------
 
-- **Search Methods**: Explores the transformation space and attempts to find a successful attack
-- **Transformations**: Takes a text input and transforms it by replacing words and phrases while attempting to retain the meaning
-- **Constraints**: Determines if a given transformation is valid
-- **Built-in Datasets** and **Pre-trained Models** for ease of use
+TextAttack isn't just a Python library; it's a framework for constructing adversarial attacks in NLP. TextAttack builds attacks from four components:
+
+- **Goal Functions** stipulate the goal of the attack, like to change the prediction score of a classification model, or to change all of the words in a translation output
```

Comment:
It would be cool if Goal Functions linked to the Goal Function documentation, etc.

Reply:
agreed, we'll add this when we add the last stuff